pkgs/tools/misc/esphome/default.nix

```
@@ -1,36 +1,61 @@
-{ lib, python3, platformio, esptool, git, protobuf3_12, fetchpatch }:
-
-let
-  python = python3.override {
-    packageOverrides = self: super: {
-      protobuf = super.protobuf.override {
-        protobuf = protobuf3_12;
-      };
-    };
-  };
+{ lib
+, python3
+, fetchFromGitHub
+, platformio
+, esptool
+, git
+}:
 
-in python.pkgs.buildPythonApplication rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "esphome";
-  version = "1.16.0";
+  version = "1.17.0";
```

Support seems to be limited to [Python > 3.7](https://github.com/esphome/esphome/blob/dev/setup.py#L77).

```suggestion
  version = "1.17.0";
  disabled = pythonOlder "3.7";
  
```

Or setting it to "3.8" to avoid issues with `asyncmock` which would probably be needed as `checkInputs` for Python 3.7 if one want to run it with a older Python release.

---

What would be the attribute to build this with another python version?